docs/ecs_guide.md

```diff
@@ -0,0 +1,269 @@
+# Deploy an ECS Worker in AWS
+
+```mermaid
```

Comment:
This pushes all the content below the fold for me — I don't think it should necessarily be at the top. I'd create a dedicated section for it "Architectural diagram" or something.